python_modules/libraries/dagster-embedded-elt/dagster_embedded_elt/sling/sling_event_iterator.py

```diff
@@ -16,20 +30,149 @@
 T = TypeVar("T", bound=SlingEventType)
 
 
+def _get_logs_for_stream(
```

Comment:
Should this be cached? Should we expect multiple table schemas to be extracted from the same stream?

Reply:
Yeah, I think it's worth caching if we add other deferred processing steps - for now should just be once per stream